matrix_client/client.py

```diff
@@ -604,9 +604,15 @@ def _sync(self, timeout_ms=30000):
             # TODO: the rest of this for loop should be in room object method
             room.prev_batch = sync_room["timeline"]["prev_batch"]
 
+            new_users = set()
```

Comment:
I was reflecting on this thinking it was ugly, and I noticed that it was also completely broken (doesn't look at the join events in `timeline`). But, this has been there for a month with everything working perfectly in all of my tests. In fact I think this is simply unneeded as the `changed` key of `device_lists` of the sync response already tells us everything we need to know ie it is populated the first time we share an encrypted room with a user. If whoever reviews this agrees, I'll drop this commit.